### PR TITLE
Louquinze development

### PR DESCRIPTION
**Pull request by mfeurer**
Follow-up on #1575

### CODE REVIEW

File: test/test_metalearning/pyMetaLearn/test_metalearning_configuration.py [hunk 1]
```diff
@@ -17,24 +17,30 @@ def test_metalearning_cs_size(self):
...
-            ({"A": "numerical", "B": "string"}, 180),
-            ({"A": "categorical", "B": "string"}, 177),
-            ({"A": "categorical", "B": "string", "C": "numerical"}, 183),
+            ({"A": "numerical"}, total - num_string - num_categorical),
+            ({"A": "categorical"}, total - num_string - num_numerical),
+            ({"A": "string"}, total - num_categorical - num_numerical),
+            ({"A": "numerical", "B": "categorical"}, total - num_string),
+            ({"A": "numerical", "B": "string"}, total - num_categorical),
+            ({"A": "categorical", "B": "string"}, total - num_numerical),
+            ({"A": "categorical", "B": "string", "C": "numerical"}, total),
```
Comment (eddiebergman) [inline test/test_metalearning/pyMetaLearn/test_metalearning_configuration.py:32]: Big fan of this change

**Review by eddiebergman — APPROVED** [review]
Looks mostly good as far as I can tell, I didn't read the test file.
One small nitpick about a comment that has an assumption which could just as well be an assert, but otherwise looks good and mostly the same.

I'll have a quick look commit by commit as well just to make it easier

File: autosklearn/pipeline/components/data_preprocessing/text_encoding/tfidf_encoding.py [hunk 2]
```diff
@@ -153,15 +157,14 @@ def get_hyperparameter_search_space(
             default_value=4,
         )
 
-        hp_ngram_range_word = CSH.UniformFloatHyperparameter(
+        hp_ngram_range_word = CSH.UniformIntegerHyperparameter(
```
Comment (eddiebergman) [inline autosklearn/pipeline/components/data_preprocessing/text_encoding/tfidf_encoding.py:160]: Good spot on making this Int

File: autosklearn/pipeline/components/data_preprocessing/text_encoding/tfidf_encoding.py [hunk 3]
```diff
@@ -7,87 +7,125 @@
...
-            else:
-                X_transformed += self.preprocessor.transform(X[feature])
-        return X_transformed
+        if self.per_column:
+            X_new = None
+            if self.preprocessor is None:
+                raise NotImplementedError()
+
+            for feature in self.preprocessor:
+                # the names in the dataframe must not change
```
[the remark this answers is not in the document]
Comment (mfeurer) [reply autosklearn/pipeline/components/data_preprocessing/text_encoding/tfidf_encoding.py:103]: This should be part of the feature validator, I opened issue #1579 for this.

**Pull request merged**